src/v/cluster/rm_stm.cc

```diff
@@ -2726,4 +2734,21 @@ ss::future<> rm_stm::clear_old_tx_pids() {
       });
 }
 
+ss::future<> rm_stm::clear_old_idempotent_pids() {
+    while (_log_state.lru_idempotent_pids.size() > 100500) {
+        auto pid_for_delete = _log_state.lru_idempotent_pids.front().entry.pid;
+        auto mutex = get_idempotent_producer_lock(pid_for_delete);
+        auto lock = mutex->try_get_units();
```

Comment:
we should take write lock and co_await until it's taken

Reply:
No, it does not work. Because we can run this code inside replicate_seq. So we will be in situation 
```
- start replicate_seq
- get read_lock (inside replicate)
- clean_old_pids
- co_await write_lock
```
Deadlock